types/header.go

```diff
@@ -112,6 +120,38 @@ func (h *Header) ValidateBasic() error {
 	return nil
 }
 
+// clockDrift defines how much new header's time can drift into
+// the future relative to the now time during verification.
+var maxClockDrift = 10 * time.Second
+
+// verifyNewHeaderAndVals performs basic verification of untrusted header.
+func verifyNewHeaderAndVals(trusted, untrusted *Header) error {
+	if err := untrusted.ValidateBasic(); err != nil {
+		return fmt.Errorf("untrusted.ValidateBasic failed: %w", err)
+	}
+
+	if untrusted.Height() <= trusted.Height() {
+		return fmt.Errorf("expected new header height %d to be greater than one of old header %d",
+			untrusted.Height(),
+			trusted.Height())
+	}
+
+	if !untrusted.Time().After(trusted.Time()) {
+		return fmt.Errorf("expected new header time %v to be after old header time %v",
+			untrusted.Time(),
+			trusted.Time())
+	}
+
+	if !untrusted.Time().Before(time.Now().Add(maxClockDrift)) {
+		return fmt.Errorf("new header has a time from the future %v (now: %v; max clock drift: %v)",
+			untrusted.Time(),
+			time.Now(),
+			maxClockDrift)
+	}
+
+	return nil
+}
+
```

Comment:
Every check in this func is now done by go-header automatically, thus is removed in main. No need rocadd it back.

Reply:
Thanks, removed.